templates/quality.html

```diff
@@ -0,0 +1,132 @@
+      </select>
+      <input type="text" class="form-control" name="stderr_value[]" value="{{ data.get('stderr_value', "") }}">
+    </div>
+  </div>
+  <div class="form-group">
+    <button type="button" class="copy-btn btn btn-outline-warning {{'d-none' if not show_add_btn}}">Add and continue on this page</button>
+    <button type="button" class="copy-btn btn btn-outline-primary {{'d-none' if not show_add_btn}}" data-submit="qa-form">Add and continue on next page</button>
+    <button type="button" data-toremove=".opt-form" class="rm-btn btn btn-outline-danger {{'d-none' if not show_rm_btn}}">Remove</button>
+  </div>
+{% endwith %}
+{%- endmacro %}
+{#- END: Quality Managment form macro -#}
+
+{#- BEGIN: Rendered content -#}
+{% block content %}
+  <script type="text/javascript">
+    $(function(){
+      $(document).on("change", ".inspect-type-checkbox", function(){
+        // Workaround: Checkboxes are only posted if checked, but we need a
+        // value per step to be able to aggregate step values by index.
+        // TODO: DRY with checkbox workaround in main.js
+        var checked = $(this).is(':checked');
+        $(this).prev("input.inspect-include").val(checked);
+
+        // Toggle inspection input fields visibility
+        // It would be nicer to disable the form inputs, but then they don't
+        // get posted which messes up the aggregation by index.
+        // Readonly would work, but only for input fields and not for select
+        $(this).parents(".inspect-type-container")
+            .find(".inspect-type-input")
+            .toggleClass("invisible");
+      });
+    });
+  </script>
+  {{ macros.progress_bar(value=40) }}
+  <hr>
+  <h1>How do you assure your software's quality?</h1>
+  <p>Do you run any static or dynamic tests? Below we provide some popular tools. Please pick the one(s) you are using, specify the executed command and tell us how you verify that the tests ran through properly.</p>
+
+  {#- BEGIN: Option Grid -#}
+  {% call(data) macros.option_grid(options=options) %}
+    {{ form_content(data={ "cmd": data.get("cmd") }, show_add_btn=True, show_rm_btn=False) }}
+  {% endcall %}
+  {#- END: Option Grid -#}
+
+  {#- BEGIN: Postable Form Content -#}
+  {% call(data=None) macros.form_container(form_id="qa-form", form_action=url_for('quality_management'), user_data=user_data) %}
+    {{ form_content(data=data) }} {#- data passed back from macro -#}
+  {% endcall %}
+  {#- END: Postable Form Content -#}
+
+  {#- BEGIN: Navigation Footer (Previous/Next) -#}
+  <hr>
+  <div class="row no-gutters justify-content-between">
+    <div class="col-3">
+      <a class="btn btn-outline-primary btn-block" href="{{ url_for('building')}} ">Previous</a>
+    </div>
+    <div class="col-3">
+      <input type="submit" form="qa-form" class="btn btn-outline-primary btn-block" value="Next">
+    </div>
+  </div>
+  {#- END: Navigation Footer (Previous/Next) -#}
+
+{% endblock %}
+{#- END: Rendered content -#}
```

Comment:
I thought many of these templates shared a lot of code. Was this the easiest way to abstract it out? Or did you separated it this way because you foresee each file changing in different ways.

Reply:
...more a question of code evolution. As you can see, I already employ various template-related DRY techniques ([macros](https://github.com/in-toto/layout-web-tool/blob/wizard-beta/templates/_macros.html) and [inheritance](https://github.com/in-toto/layout-web-tool/blob/wizard-beta/templates/base.html)), but I agree that there is room for more.

Reply:
Will fix with https://github.com/in-toto/layout-web-tool/issues/32